src/js_mixed.hpp

```diff
@@ -0,0 +1,187 @@
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleString : public MixedWrapper<Context, Value> {
+   private:
+    // we need this <cache> to keep the value life long enough to get into the
+    // DB, we do this because the realm::Mixed type is just a reference
+    // container.
+    std::string cache;
+
+   public:
+    Mixed wrap(Context context, Value const &value) {
+        cache = Utils::to_string(context, value);
+        return Mixed(cache);
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_string(context,
+                                  mixed.template get<StringData>().data());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleBoolean : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        return Mixed(Utils::to_boolean(context, value));
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_boolean(context, mixed.get<bool>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils,
+          typename RealmNumberType>
+class HandleNumber : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        return Mixed(Utils::to_number(context, value));
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_number(context, mixed.get<RealmNumberType>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleDecimal128 : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        return Mixed(Utils::to_decimal128(context, value));
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_decimal128(context, mixed.get<Decimal>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleObjectID : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        return Mixed(Utils::to_object_id(context, value));
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_object_id(context, mixed.get<ObjectId>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleBinary : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        auto owned_binary_data = Utils::to_binary(context, value);
+        return Mixed(owned_binary_data.get());
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_binary(context, mixed.get<BinaryData>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleTimeStamp : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        auto date = Utils::to_date(context, value);
+
+        double milliseconds = Utils::to_number(context, date);
```

Comment:
[q] Isn't it better to `std::move` the `date` (and the `ts` at the end) variables here?